src/v2/api/services/ScheduleService.ts

```diff
@@ -237,4 +244,175 @@ export class ScheduleService {
       discipline: result.discipline,
     };
   }
+
+  private async getEventInfoForUpdate (
+    period: Period,
+    startTime: Date,
+    endTime: Date,
+    week: number,
+    pairDescription: string,
+    event: DbEvent,
+  ) {
+    const eventInfo = {
+      update: undefined,
+      createMany: undefined,
+      deleteMany: undefined,
+    };
+
+    if (pairDescription) {
+      const { endOfWeek } = await this.dateService.getDatesOfWeek(week);
+      const indexOfLesson = this.getIndexOfLesson(period, startTime, endOfWeek);
+      if (indexOfLesson === null || indexOfLesson > event.eventInfo.length-1) {
+        throw new InvalidWeekException();
+      }
+      eventInfo.update = {
+        where: {
+          eventId_number: {
+            eventId: event.id,
+            number: indexOfLesson,
+          },
+        },
+        data: {
+          description: pairDescription,
+        },
+      };
+    }
+
+    if (period === Period.NO_PERIOD) {
+      eventInfo.deleteMany = {
+        number: {
+          gte: 1,
+        },
+      };
+    } else {
+      const countOfWeek = Math.ceil((endTime.getTime() - startTime.getTime()) / WEEK);
+      const countOfPair = period === Period.EVERY_FORTNIGHT ? Math.floor(countOfWeek/2) : countOfWeek;
+      const pairDifference = countOfPair - event.eventInfo.length;
+
+      if (pairDifference > 0) {
+        eventInfo.createMany = {
+          data: this.scheduleParser.getIndexesForEventInfo(event.eventInfo.length, event.eventInfo.length+pairDifference-1),
+        };
+      } else if (pairDifference < 0) {
+        eventInfo.deleteMany = {
+          number: {
+            gte: countOfPair-1,
+          },
+        };
+      }
+    }
+    return eventInfo;
+  }
+
+  async getDisciplineForUpdate (
+    type: DisciplineTypeEnum,
+    disciplineTypeId: string,
+    disciplineDescription: string,
+    teachers: string[],
+    disciplineTeachers: DbDisciplineTeacher[],
+  ) {
+    const update = {
+      description: undefined,
+      disciplineTypes: {},
+      disciplineTeachers: {
+        deleteMany: {
+          OR: [],
+        },
+        createMany: {
+          data: [],
+        },
+      },
+    };
+
+    update.description = disciplineDescription ? disciplineDescription : undefined;
+
+    if (type) {
+      update.disciplineTypes = {
```

Comment:
It's not correct. Changing the type doesn't change the discipline type entity. It disconnects and connects another discipline type